typescript/cli/src/send/transfer.ts

```diff
@@ -179,74 +130,77 @@ async function executeDelivery({
+      return;
     }
+
+    routerAddress = (await select({
+      message: `Select router address`,
+      choices: [
+        ...tokensForRoute.map((t) => ({
+          value: t.addressOrDenom,
+        })),
+      ],
+      pageSize: 10,
+    })) as string;
   }
 
-  let adapter: IHypTokenAdapter<PopulatedTransaction>;
-  const multiProtocolProvider =
-    MultiProtocolProvider.fromMultiProvider(multiProvider);
-  if (tokenType === TokenType.native) {
-    adapter = new EvmHypNativeAdapter(origin, multiProtocolProvider, {
-      token: routerAddress,
-    });
-  } else if (tokenType === TokenType.collateral) {
-    adapter = new EvmHypCollateralAdapter(origin, multiProtocolProvider, {
-      token: routerAddress,
-    });
-  } else {
-    adapter = new EvmHypSyntheticAdapter(origin, multiProtocolProvider, {
-      token: routerAddress,
-    });
+  const token = warpCore.findToken(origin, routerAddress);
+  if (!token) {
+    logRed(
+      `No Warp Routes found from ${origin} to ${destination} with router address ${routerAddress}`,
+    );
+    return;
+  }
+
+  const senderAddress = await signer.getAddress();
+  const errors = await warpCore.validateTransfer({
+    originTokenAmount: new TokenAmount(wei, token),
+    destination,
+    recipient: recipient ?? senderAddress,
+    sender: senderAddress,
+  });
+  if (errors) {
+    logRed('Unable to validate transfer');
+    return;
+  }
+
+  const transferTxs = await warpCore.getTransferRemoteTxs({
+    originTokenAmount: new TokenAmount(wei, token),
+    destination,
+    sender: senderAddress,
+    recipient: recipient ?? senderAddress,
+  });
+
+  const txReceipts = [];
+  for (const tx of transferTxs) {
+    if (tx.type === ProviderType.EthersV5) {
+      const txResponse = await connectedSigner.sendTransaction(tx.transaction);
+      const txReceipt = await multiProvider.handleTx(origin, txResponse);
+      txReceipts.push(txReceipt);
+    }
   }
 
-  const destinationDomain = multiProvider.getDomainId(destination);
-  log('Fetching interchain gas quote');
-  const interchainGas = await adapter.quoteTransferRemoteGas(destinationDomain);
-  log('Interchain gas quote:', interchainGas);
-  const transferTx = (await adapter.populateTransferRemoteTx({
-    weiAmountOrId: wei,
-    destination: destinationDomain,
-    recipient,
-    interchainGas,
-  })) as ethers.PopulatedTransaction;
-
-  const txResponse = await connectedSigner.sendTransaction(transferTx);
-  const txReceipt = await multiProvider.handleTx(origin, txResponse);
-
-  const message = core.getDispatchedMessages(txReceipt)[0];
+  const transferTxReceipt = txReceipts[txReceipts.length - 1];
+
+  const message = core.getDispatchedMessages(transferTxReceipt)[0];
   logBlue(`Sent message from ${origin} to ${recipient} on ${destination}.`);
   logBlue(`Message ID: ${message.id}`);
+  logBlue(
+    `View on Hyperlane Explorer: https://explorer.hyperlane.xyz/message/${message.id}`,
```

Comment:
This unfortunately won't work for PI chains the explorer doesn't know about. Users would need to add their custom chain metadata there first. It's possible to include chainmetadata along with the msg id but don't feel like you have to tackle that right now:
https://github.com/hyperlane-xyz/hyperlane-explorer/blob/main/src/features/chains/useChainConfigs.ts#L34

Reply:
ahh good point, I'll remove this line and leave it for now then